graphbolt/src/cuda/neighbor_sampler.cu

```diff
@@ -65,7 +65,8 @@ __global__ void _ComputeRandoms(
     }
 
     const auto rnd = curand_uniform(&rng);
-    const auto prob = weights ? weights[in_idx] : static_cast<weights_t>(1);
+    const auto prob =
+        sliced_weights ? sliced_weights[i] : static_cast<weights_t>(1);
```

Comment:
if `all(probs == 0)`, then `sliced_weights` is nullptr and all edges will be candidates for sampling? equivalent to `probs_or_mask` is `None` in `SampleNeighbors()`?

Reply:
Yes, when `probs_or_mask` is none, we assume the probabilities are equal.

Reply:
However, if we pass `probs_or_mask` filled with entries `0`, then no edges will be sampled.

Reply:
actually my question is should `probs_or_mask=None` be same as `probs=_or_mask=torch.zeros()`?

Reply:
No

Reply:
In that case, no edges will be sampled.

Reply:
`probs_or_mask=torch.ones()` is equivalent to having `probs_or_mask=None`